tmt/convert.py

```diff
@@ -109,47 +110,80 @@ def read(path, makefile, nitrate, purpose):
-        # Contact
-        if testcase.tester:
-            data['contact'] = '{} <{}>'.format(
-                testcase.tester.name, testcase.tester.email)
-            echo(style('contact: ', fg='green') + data['contact'])
-        # Environment
-        if testcase.arguments:
-            data['environment'] = tmt.utils.variables_to_dictionary(
-                testcase.arguments)
-            echo(style('environment:', fg='green'))
-            echo(pprint.pformat(data['environment']))
-        # Relevancy
-        field = tmt.utils.StructuredField(testcase.notes)
-        data['relevancy'] = field.get('relevancy')
-        echo(style('relevancy:', fg='green'))
-        echo(data['relevancy'].rstrip('\n'))
+            log.warn("Multiple test cases found for '{0}'.".format(
+                test))
+
+        testcase_data = list()
+
+        for testcase in testcases:
+            single_case_data = dict()
+            echo("test case found '{0}'.".format(testcase.identifier))
+            # Test identifier
+            single_case_data['tcms id'] = testcase.identifier
+            # Test name
+            if testcase.summary:
+                single_case_data['testname'] = '{}'.format(
+                    testcase.summary)
+                echo(style('test name: ', fg='green') + single_case_data['testname'])
+            # Contact
+            if testcase.tester:
+                single_case_data['contact'] = '{} <{}>'.format(
+                    testcase.tester.name, testcase.tester.email)
+                echo(style('contact: ', fg='green') + single_case_data['contact'])
+            # Environment
+            if testcase.arguments:
+                single_case_data['environment'] = tmt.utils.variables_to_dictionary(
+                    testcase.arguments)
+                echo(style('environment:', fg='green'))
+                echo(pprint.pformat(single_case_data['environment']))
+            # Relevancy
+            field = tmt.utils.StructuredField(testcase.notes)
+            single_case_data['relevancy'] = field.get('relevancy')
+            echo(style('relevancy:', fg='green'))
+            echo(single_case_data['relevancy'].rstrip('\n'))
+            testcase_data.append(single_case_data)
+
+        common_candidates = dict()
+
+        # Find common data from individual test cases
+        for testcase in testcase_data:
+            if testcase_data.index(testcase) == 0:
+                common_candidates = copy.copy(testcase)
+            else:
+                for key, value in testcase.items():
+                    if key in common_candidates:
+                        if value != common_candidates[key]:
+                            common_candidates.pop(key)
+
+        # Add common data to main.fmf
+        for key, value in common_candidates.items():
+            data[key] = value
+
+        # Remove common data from individual fmfs
+        for common_key in list(common_candidates):
```

Comment:
I guess this could be just `for common_key in common_candidates`, right?

Reply:
I was convinced that without the `list()` it caused some kind of problem but I cannot replicate it now so I guess it is pointless.